chore/release-beta.sh

```diff
@@ -19,17 +19,45 @@ echo "NEW_BETA_NUMBER $NEW_BETA_NUMBER"
 NEXT_VERSION="$CURRENT_VERSION-beta-$NEW_BETA_NUMBER"
 echo "::endgroup::"
 
-echo "::group::Setting release version"
+BRANCH_NAME="beta-release/$NEXT_VERSION"
+
+echo "::group::Setting beta-release version"
 mvn -f spoon-pom --no-transfer-progress --batch-mode versions:set -DnewVersion="$NEXT_VERSION" -DprocessAllModules
 mvn --no-transfer-progress --batch-mode versions:set -DnewVersion="$NEXT_VERSION" -DprocessAllModules
 mvn -f spoon-javadoc --no-transfer-progress --batch-mode versions:set -DnewVersion="$NEXT_VERSION" -DprocessAllModules
 echo "::endgroup::"
 
-echo "::group::Staging release"
+echo "::group::Commit & Push changes"
+git checkout -b "$BRANCH_NAME"
+git commit -am "release: Releasing version $NEXT_VERSION"
+git push --set-upstream origin "$BRANCH_NAME"
+echo "::endgroup::"
+
+echo "::group::Staging beta-release"
 mvn -f spoon-pom --no-transfer-progress --batch-mode -Pjreleaser clean deploy -DaltDeploymentRepository=local::default::file:./target/staging-deploy
 mvn --no-transfer-progress --batch-mode -Pjreleaser deploy:deploy-file -Dfile="./spoon-pom/pom.xml" -DpomFile="./spoon-pom/pom.xml" -Durl="file://$(mvn help:evaluate -D"expression=project.basedir" -q -DforceStdout)/target/staging-deploy"
 echo "::endgroup::"
 
 echo "::group::Running jreleaser"
-JRELEASER_PROJECT_VERSION="$NEXT_VERSION" jreleaser-cli deploy
+JRELEASER_PROJECT_VERSION="$NEXT_VERSION" jreleaser-cli release
+echo "::endgroup::"
+
+# Set next version (patch of release version) with -SNAPSHOT suffix
+NEXT_RELEASE_VERSION=$CURRENT_VERSION_WITH_SNAPSHOT
+
+echo "::group::Updating poms to next target version"
+mvn -f spoon-pom --no-transfer-progress --batch-mode versions:set -DnewVersion="$NEXT_RELEASE_VERSION" -DprocessAllModules
+mvn --no-transfer-progress --batch-mode versions:set -DnewVersion="$NEXT_RELEASE_VERSION" -DprocessAllModules
+mvn -f spoon-javadoc --no-transfer-progress --batch-mode versions:set -DnewVersion="$NEXT_RELEASE_VERSION" -DprocessAllModules
+echo "::endgroup::"
+
+echo "::group::Committing changes"
+git commit -am "release: Setting SNAPSHOT version back to $NEXT_RELEASE_VERSION"
+git push --set-upstream origin "$BRANCH_NAME"
+echo "::endgroup::"
+
+echo "::group::Merging into master (fast-forward)"
+git checkout master
+git merge --ff-only "$BRANCH_NAME"
+git push origin master
```

Comment:
Which means I don't quite get what change you actually committed and are pushing back here.